sdk/core/azure-mgmt-core/azure/mgmt/core/_pipeline_client.py

```diff
@@ -52,13 +52,13 @@ def __init__(self, base_url, **kwargs):
                 )
             per_call_policies = kwargs.get('per_call_policies', [])
             if isinstance(per_call_policies, Iterable):
-                per_call_policies.append(ARMAutoResourceProviderRegistrationPolicy())
+                per_call_policies = list(per_call_policies).append(ARMAutoResourceProviderRegistrationPolicy())
```

Comment:
Fix https://github.com/Azure/azure-sdk-for-python/pull/18238#discussion_r620131467

`append` will fail if `per_call_policies` is a `tuple`, instead of a `list`:

> AttributeError: 'tuple' object has no attribute 'append'

Convert to `per_call_policies` to a `list` first.